bin/diagnostic.js

```diff
@@ -1 +1,20 @@
 'use strict';
+
+
+
+//We'll learn about require later in the course
+const ask = require('../lib/ask.js');
+
+let answer = NaN;
+
+while (answer !== NaN) {
```

Comment:
Your code works perfectly, but you should know that you should not be checking the opposite of the initial value of "answer" for this while loop. The reason your code is working is actually because of a strange (and frankly, totally illogical) behavior in Javascript whereby NaN === NaN will evaluate to false. (Yea, I know!)

If you had written, "let answer = 'hello'" and "while (answer !== 'hello')", your loop would never run. Because answer is assigned the value "hello", and the while loop checks whether it is NOT "hello" before there is any opportunity for the value of answer to be changed.